### ISSUE TITLE
UI/UX

### ISSUE THREAD

**Issue by AddisonFreeman** [robgpeak/CL-WebExtension#7]
@robgpeak I just pushed a first pass at updating the style. I hardcoded the cursive username tab but that will dropdown with an animation (I'll need to pull that data from the API *nudge* @cgladue *nudge*). Next I will include the points as a dashboard like ebates does. Now for news I'm not getting any for my account, afree1423@gmail.com, can you change that in the backend so I can style the newsfeed too? Thanks. We can also talk about different themes for each subdomain and since I'm using a design tool to style the extension it would be easy to have this view as a standalone webpage/mobile page. 

In summary, points for discussion:
-animations
-fonts
-newsfeed
-per subdomain theming
-standalone webpage for mobile

<img width="410" alt="screen shot 2018-03-26 at 4 30 05 pm" src="https://user-images.githubusercontent.com/5950267/37934307-6735d546-3113-11e8-99a2-191c3e97a60a.png">

**Comment by cgladue** [comment]
I will try to set that flag on the cookie see if it makes a difference but other than that I can’t and would like to have some suggestions from you and what would be best for you and I deal situation and how do identify that

**Comment by robgpeak** [comment]
Just so you know, I am not really digging the cursive font.  Looking for something normal and easier to read.

**Comment by AddisonFreeman** [comment]
@cgladue Is it possible to get the colors in hex or rbg format? Here's the primary blue and secondary orange for complinks in hex: #56B5E4, #FF8C00

**Comment by AddisonFreeman** [comment]
@cgladue Also, xclub's partnerStyle field is invalid JSON, there's trailing commas.

**Comment by cgladue** [comment]
No I can’t without adding them in the style json. The front end uses angular and whatever format that it’s in currently.

**Comment by cgladue** [comment]
I’ll take a look and fix it

**Comment by AddisonFreeman** [comment]
Could you add a hex or rgb field then?

**Comment by cgladue** [comment]
do you know how to convert those values to hex ?



Chad Gladue
*DevNull Technology, LLC*

**Comment by AddisonFreeman** [comment]
I found this color map which I may be able to use to convert:
https://github.com/mui-org/material-ui/blob/master/src/styles/colors.js.
This trailing comma in the partnerStyle field is preventing me from using
anything though, can you fix that real quick and I'll see if I can work
with the existing format?

**Comment by AddisonFreeman** [comment]
<img width="708" alt="screen shot 2018-03-30 at 1 13 05 pm" src="https://user-images.githubusercontent.com/5950267/38148495-5d811658-341c-11e8-905b-891e1ba8f20a.png">
@cgladue

**Comment by cgladue** [comment]
should be fixed now.



Chad Gladue
*DevNull Technology, LLC*

**Comment by AddisonFreeman** [comment]
@robgpeak go ahead and check the notification and popup styling for multidomain. I'll finish checkin/checkout on the settings page by EOD, for now still login and out from the shop.complinks and xclub.complinks sites

<img width="635" alt="screen shot 2018-04-02 at 12 46 36 pm" src="https://user-images.githubusercontent.com/5950267/38207561-eaf28180-3673-11e8-9974-d8ad63443413.png">
<img width="749" alt="screen shot 2018-04-02 at 12 46 24 pm" src="https://user-images.githubusercontent.com/5950267/38207562-eb04bd6e-3673-11e8-97d1-723c245cdb85.png">
<img width="651" alt="screen shot 2018-04-02 at 12 47 24 pm" src="https://user-images.githubusercontent.com/5950267/38207604-0892a95e-3674-11e8-8179-bae621f177ac.png">
<img width="638" alt="screen shot 2018-04-02 at 12 47 17 pm" src="https://user-images.githubusercontent.com/5950267/38207605-08a3c9a0-3674-11e8-86d0-305a16fc9b0c.png">

**Comment by robgpeak** [comment]
I can't look at it right now, working on an RFP due tomorrow, but two things visually.  

1. Make the corners of on page notification radius like you did with the overall bigger extension window.
2. Make all the font a little smaller so the "Click here to earn 0 points/$1 spent" fits in one line, also taking into account that the "0" could be up to 3 characters in length.

Lastly, we need to show the user that they have in fact activated the "trip" by showing them a notification like everyone else does...  So it would look something like this...
<img width="438" alt="screen shot 2018-04-02 at 1 57 41 pm" src="https://user-images.githubusercontent.com/28393235/38208083-dcf0df50-367d-11e8-90b9-683325458bbe.png">

**Comment by robgpeak** [comment]
Yes that makes sense.  Great idea Chad!

**Comment by robgpeak** [comment]
@AddisonFreeman We need to make the image above the listing smaller.  

1. Let's constrain the size to sort of the image 25px high.  
2. The same thing with the button. Let's constrain that to be 25px high as well.
3. Make the font in the button a little smaller.
4. Change the text on the button to read, "Activate xx points/$1 spent" (of course leaving room for the "up to" in between 'Activate and xx' if necessary.
5. Stop our image and the activate buttons from showing up in the "sub links" to the same site. (see image below.

<img width="1155" alt="screen shot 2018-04-12 at 2 48 27 pm" src="https://user-images.githubusercontent.com/28393235/38697752-e06aceee-3e60-11e8-9cb2-c8adf22f3ab6.png">

**Comment by AddisonFreeman** [comment]
<img width="710" alt="screen shot 2018-04-14 at 5 00 43 pm" src="https://user-images.githubusercontent.com/5950267/38773608-9a3c1c1a-4005-11e8-9836-104946b158ea.png">
<img width="660" alt="screen shot 2018-04-14 at 5 00 27 pm" src="https://user-images.githubusercontent.com/5950267/38773609-9a82fb08-4005-11e8-892a-d73733da6873.png">
Made these changes and pushed the new code.

**Comment by robgpeak** [comment]
Looks good.  Can you do me a favor and make the activate button a little smaller in height... Maybe 4 or 6 pixels smaller.

Then how/when will it start working on all the advertisers extension enabled true.

**Comment by AddisonFreeman** [comment]
That's on Chad's end.

**Comment by robgpeak** [comment]
@cgladue can you please do that to the height of the button.  Please reduce it by like 4 or 6 pixels.

**Comment by AddisonFreeman** [comment]
Ill fix the buttons. the enabled for all sites is on the backend

**Comment by AddisonFreeman** [comment]
Pushed the -5 pixels fix.

**Comment by AddisonFreeman** [comment]
Just pushed the google result activated notification showing that the deal
was activated. This should mark phase 2 complete from my end.

**Comment by AddisonFreeman** [comment]
I've reverted those changes. This should mark phase 2 complete.

**Comment by robgpeak** [comment]
Do I need to install a new version to see those changes?

**Comment by AddisonFreeman** [comment]
Yes

**Comment by robgpeak** [comment]
It is also not working globally on all google search results that show our advertisers who from what I see in our back end are extension enabled "true".  Meaning, if I do a search for Best Buy, I see our graphic and activate link above the listing, but not for any other sites that I search for who are, at least from what I can tell, extension enabled?

**Comment by AddisonFreeman** [comment]
I pushed, sorry about that. Does the notification appear if you click on
that page link?

**Comment by robgpeak** [comment]
Addison, please get me on skype because we must be looking at two different things.  It is not working for all sites that are marked in the back end as extension enabled, and it is still launching a shopping trip directly fro the Activate link in the button next to the listing.  Here is a screencast of what I am seeing, and what I would like to be seeing...  https://screencast.com/t/NlWzUU3iE1

**Comment by AddisonFreeman** [comment]
I pushed again (my fault for not catching the gist), 1.3.6 that doesn't not
start a trip.

For the second issue:

This is the logic that determines if a google result should have the
activate link: "if(data.isAdvertiser && data.extensionEnabled)"

it is the exact same logic that triggers the notification on a page.

If you click on shoes.com and see a notification appear and do not see a
link for that site on the google result page, then let me know, but as far
as I can tell this is in Chad's realm, I believe the api is not connected
to the backend of what you're seeing.

**Comment by robgpeak** [comment]
@cgladue It is that time for a screencast.  This might be two separate issues, but I am going to show them in one screen cast.

1. The advertisers in the back end that are showing extension Enabled are NOT appearing in the search results on google with our link above them.  Best Buy is, but that is the only one.

2. It seems like it take a long time for our link to show up compared to the southwest and ebates extensions...  https://screencast.com/t/1tuioykKx

**Comment by cgladue** [comment]
if the extension is not enabled then shouldnt it not appear in the search
results ?  thats part of the extension

as for the speed i dont know why that is, the API call takes less then half
a second to complete ~217 milliseconds  addison how are you getting the
results for all the results\, you firing one request for each result on the
page ?



Chad Gladue
*DevNull Technology, LLC*

**Comment by cgladue** [comment]
Macys isnt showing up because there is no "Program URL" configured for the
advertiser ... this field needs to be filled in with the website name (i.e.
http://www.macys.com)





Chad Gladue
*DevNull Technology, LLC*

**Comment by cgladue** [comment]
It may be laborious but it’s only thing I can do if they advertise it does not send a crossed it looks like Link share is the only one that doesn’t send over the URL commission Junction should feel it in automatically

**Comment by AddisonFreeman** [comment]
Chad I am doing one request at a time for each search result. Is it
possible to do a list of domains in one xhr?

**Comment by cgladue** [comment]
yea i was thinking of allowing that, let me throw something together.



Chad Gladue
*DevNull Technology, LLC*

**Comment by cgladue** [comment]
@robgpeak @AddisonFreeman 

the checkDomain call for both macys and godaddy seems to be working fine from my side ...

```
{
  "domainName" : "godaddy.com"
}

{
    "isAdvertiser": true,
    "extensionEnabled": true,
    "advertiserName": "GoDaddy.com",
    "clickUrl": "https://shop.complinks.co/trip/start?aid=11&lid=0&url=http%3a%2f%2fwww.kqzyfj.com%2fclick-8235261-10378406-1491592212000",
    "logoUrl": "https://s3.us-east-2.amazonaws.com/complinks/logos/1513033.jpg",
    "reward": "up to 7.5 points/$1 spent"
}

{
  "domainName" : "macys.com"
}

{
    "isAdvertiser": true,
    "extensionEnabled": true,
    "advertiserName": "Macys.com",
    "clickUrl": "https://shop.complinks.co/trip/start?aid=6789&lid=0&url=https%3a%2f%2fclick.linksynergy.com%2ffs-bin%2fclick%3fid%3djLQ8qrFPMSA%26offerid%3d486109.100258994%26type%3d3",
    "logoUrl": "https://s3.us-east-2.amazonaws.com/complinks/logos/lg_3184.jpg",
    "reward": "1.5 points/$1 spent"
}

```

**Comment by AddisonFreeman** [comment]
<img width="1413" alt="screen shot 2018-04-17 at 3 26 45 pm" src="https://user-images.githubusercontent.com/5950267/38902092-da5d49ea-4253-11e8-8e9a-0893df688b45.png">
not so on my account. afree1423@gmail.com

**Comment by AddisonFreeman** [comment]
<img width="1415" alt="screen shot 2018-04-17 at 3 28 11 pm" src="https://user-images.githubusercontent.com/5950267/38902112-f8966b8a-4253-11e8-8814-cd53f68ce29a.png">
on either subdomain

**Comment by cgladue** [comment]
i have this fixed on test.complinks.co  just waiting for rob to give the OK to move it over ... also i created that new call to submit multiple domains at once and ill return you a response with the status of each.

New checkDomain Request:
```
{
	"domainName" : [ 
		"macys.com",
		"godaddy.com",
		"vistaprint.com",
		"chadgladue.com"
		]
}
```

new Response:

```
[
    {
        "isAdvertiser": true,
        "extensionEnabled": true,
        "advertiserName": "Macys.com",
        "clickUrl": "https://shop.complinks.cohttps%3a%2f%2fclick.linksynergy.com%2ffs-bin%2fclick%3fid%3djLQ8qrFPMSA%26offerid%3d486109.100258994%26type%3d3",
        "logoUrl": "https://s3.us-east-2.amazonaws.com/complinks/logos/lg_3184.jpg",
        "reward": "1.5 points/$1 spent",
        "domainName": "macys.com",
        "availablePointBal": 7161,
        "pendingPointBal": 0
    },
    {
        "isAdvertiser": true,
        "extensionEnabled": true,
        "advertiserName": "GoDaddy.com",
        "clickUrl": "https://shop.complinks.cohttp%3a%2f%2fwww.kqzyfj.com%2fclick-8235261-10378406-1491592212000",
        "logoUrl": "https://s3.us-east-2.amazonaws.com/complinks/logos/1513033.jpg",
        "reward": "up to 7.5 points/$1 spent",
        "domainName": "godaddy.com",
        "availablePointBal": 7161,
        "pendingPointBal": 0
    },
    {
        "isAdvertiser": true,
        "extensionEnabled": true,
        "advertiserName": "Vistaprint",
        "clickUrl": "https://shop.complinks.cohttp%3a%2f%2fwww.tkqlhce.com%2fclick-8235261-11636757-1497297572000",
        "logoUrl": "https://s3.us-east-2.amazonaws.com/complinks/logos/769194.gif",
        "reward": "3 points/$1 spent",
        "domainName": "vistaprint.com",
        "availablePointBal": 7161,
        "pendingPointBal": 0
    },
    {
        "isAdvertiser": false,
        "extensionEnabled": false,
        "advertiserName": null,
        "clickUrl": null,
        "logoUrl": null,
        "reward": null,
        "domainName": null,
        "availablePointBal": 0,
        "pendingPointBal": 0
    }
]
```

**Comment by robgpeak** [comment]
@cgladue Considering the other issue of the referring URL that we were testing on test.complinks.co which did not work out until we talk with Chris, how does that affect what we are working on in this issue with Addison?  I don't want the other issue to slow the progress of the extension down.

**Comment by cgladue** [comment]
I cannot put the code into production until they are both resolved because the files that are involved are one in the same.   So it is holding it up.

**Comment by cgladue** [comment]
@robgpeak @AddisonFreeman 

this is now live, you can use it

**Comment by robgpeak** [comment]
@AddisonFreeman Please let me know when this is done where I can do a search on google and see all of our extension enabled advertisers showing up with our logo and button above their listing?

**Comment by cgladue** [comment]
{
  "domainName" : [ "macys.com", "godaddy.com" ]
}


[
    {
        "isAdvertiser": true,
        "extensionEnabled": true,
        "advertiserName": "Macys.com",
        "clickUrl": "https://shop.complinks.co/trip/start?aid=6789&lid=0&url=https%3a%2f%2fclick.linksynergy.com%2ffs-bin%2fclick%3fid%3djLQ8qrFPMSA%26offerid%3d486109.100258994%26type%3d3",
        "logoUrl": "https://s3.us-east-2.amazonaws.com/complinks/logos/lg_3184.jpg",
        "reward": "1.5 points/$1 spent",
        "domainName": "macys.com"
    },
    {
        "isAdvertiser": true,
        "extensionEnabled": true,
        "advertiserName": "GoDaddy.com",
        "clickUrl": "https://shop.complinks.co/trip/start?aid=11&lid=0&url=http%3a%2f%2fwww.kqzyfj.com%2fclick-8235261-10378406-1491592212000",
        "logoUrl": "https://s3.us-east-2.amazonaws.com/complinks/logos/1513033.jpg",
        "reward": "up to 7.5 points/$1 spent",
        "domainName": "godaddy.com"
    }
]

**Comment by AddisonFreeman** [comment]
The old call still works though, right? The rest of the extension uses it.

**Comment by cgladue** [comment]
no the call changed its signature



Chad Gladue
*DevNull Technology, LLC*

**Comment by AddisonFreeman** [comment]
I can change it to always send an array of one or more strings but it's
always nice when updates are backwards compatible when possible.

**Comment by cgladue** [comment]
yea once its live any changes would be...  its still in development so its
not really a concern at the moment



Chad Gladue
*DevNull Technology, LLC*

**Comment by AddisonFreeman** [comment]
Also, make sure my email is working for other sites than bestbuy.com (
afree1423@gmail.com)

**Comment by cgladue** [comment]
its not on an individual user account level its based on subdomain

**Comment by AddisonFreeman** [comment]
I'm pretty sure shop.complinks.co is still using the old call. I get a 200 response for both, but the array of strings gives false on extensionEnabled and isAdvertiser: true
not working:
<img width="1093" alt="screen shot 2018-04-19 at 2 34 39 pm" src="https://user-images.githubusercontent.com/5950267/39019689-04f7e156-43df-11e8-8e43-017314648891.png">
<img width="1095" alt="screen shot 2018-04-19 at 2 34 33 pm" src="https://user-images.githubusercontent.com/5950267/39019691-051be592-43df-11e8-96be-cb2122ab16bb.png">
working:
<img width="1091" alt="screen shot 2018-04-19 at 2 33 20 pm" src="https://user-images.githubusercontent.com/5950267/39019704-0e01479c-43df-11e8-95cd-13bacbe089cb.png">
